scipy/stats/morestats.py

```diff
@@ -1222,12 +1222,33 @@ def shapiro(x, a=None, reta=False):
     See Also
     --------
     anderson : The Anderson-Darling test for normality
+    kstest : The Kolmogorov-Smirnov test for goodness of fit.
 
     References
     ----------
     .. [1] http://www.itl.nist.gov/div898/handbook/prc/section2/prc213.htm
+    .. [2] Shapiro, S. S. & Wilk, M.B (1965). An analysis of variance test for
+           normality (complete samples), Biometrika, Vol. 52, pp. 591-611.
+    .. [3] Razali, N. M. & Wah, Y. B. (2011) Power comparisons of Shapiro-Wilk,
+           Kolmogorov-Smirnov, Lilliefors and Anderson-Darling tests, Journal of
+           Statistical Modeling and Analytics, Vol. 2, pp. 21-33.
+    .. [4] ALGORITHM AS R94 APPL. STATIST. (1995) VOL. 44, NO. 4.
+
+    Examples
+    --------
+    >>> from scipy import stats
+    >>> np.random.seed(12345678)
+    >>> x = stats.norm.rvs(loc=5, scale=3, size=100)
+    >>> stats.shapiro(x)
+    (0.9772805571556091, 0.08144091814756393)
 
     """
+    x = np.ravel(x)
+    if x.dtype == object:
+        # np.ravel and np.ndarray.flatten don't work with dtype=object
+        # i.e. they will fail to flatten if there are different sized arrays.
+        x = np.hstack(x)
```

Comment:
``[[1, 2], [3, 4], [5]]`` isn't valid array-like input. I'd expect an error here. The error is quite non-descriptive, but I don't really care too much about that. I'd prefer not to add a lot of code checking for user errors like this.

If ``asarray`` would grow some kind of argument to raise an error instead of converting to object arrays, then we could get a decent error message in a concise way. But that doesn't exist yet.